pkg/cloud/tags/tags_test.go

```diff
@@ -94,3 +131,121 @@ func TestTags_ComputeDiff(t *testing.T) {
 		})
 	}
 }
+
+func TestTags_EnsureWithEC2(t *testing.T) {
+	tests := []struct {
+		name    string
+		builder Builder
+		expect  func(m *mock_ec2iface.MockEC2APIMockRecorder)
+	}{
+		{
+			name:    "Should return error when applyFunc params fails",
+			builder: Builder{params: &bp},
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.CreateTags(gomock.Eq(&ec2.CreateTagsInput{
+					Resources: aws.StringSlice([]string{""}),
+					Tags:      tags,
+				})).Return(nil, errors.New("failed to create tag"))
+			},
+		},
+		{
+			name:    "Should return error when applyFunc is nil",
+			builder: Builder{params: &bp, applyFunc: nil},
+		},
+		{
+			name:    "Should return error when build params is nil",
+			builder: Builder{params: nil},
+		},
+		{
+			name:    "Should ensure tags successfully",
+			builder: Builder{params: &bp},
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.CreateTags(gomock.Eq(&ec2.CreateTagsInput{
+					Resources: aws.StringSlice([]string{""}),
+					Tags:      tags,
+				})).Return(nil, nil)
+			},
+		},
+	}
+
+	g := NewWithT(t)
+	mockCtrl := gomock.NewController(t)
+	ec2Mock := mock_ec2iface.NewMockEC2API(mockCtrl)
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var builder *Builder
+			if tc.expect != nil {
+				tc.expect(ec2Mock.EXPECT())
+				builder = New(tc.builder.params, WithEC2(ec2Mock))
+			} else {
+				builder = New(tc.builder.params, func(builder *Builder) {})
+			}
+			err := builder.Ensure(nil)
+			if err != nil {
+				g.Expect(err).To(Not(BeNil()))
+			} else {
+				g.Expect(err).To(BeNil())
+			}
+		})
+	}
+}
+
+func TestTags_EnsureWithEKS(t *testing.T) {
+	tests := []struct {
+		name    string
+		builder Builder
+		expect  func(m *mock_eksiface.MockEKSAPIMockRecorder)
+	}{
+		{
+			name:    "Should return error when applyFunc params fails",
```

Comment:
What does this mean?
Maybe "Should return error when tags apply function fails"?

Using `applyFunc` in test names is not descriptive enough.

Reply:
will add appropriate names

Reply:
Done